### PR TITLE
fix: push log into Sentry when US user is detected

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

To not loose the functionality with US users tradable assets restrictions, we should add logs into Sentry.

  # To Test

1. Open page with US IP
- [ ] An error `USUserError` should be logged into Sentry
- [ ] The error should contain your IP

### CODE REVIEW

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts [hunk 1]
```diff
@@ -80,12 +81,24 @@ export function TokensListsUpdater({ chainId: currentChainId }: TokensListsUpdat
...
+            message: '',
+            name: 'USUserError',
+          })
+
+          Sentry.captureException(sentryError, {
+            tags: {
+              errorType: 'USUserError',
+            },
+            contexts: { ip },
+          })
```
Comment (alfetopito) [inline libs/tokens/src/updaters/TokensListsUpdater/index.ts:101]: Should this be in Sentry, though?
Feels more like analytics than an error.

Maybe we want to capture a message instead? https://docs.sentry.io/platforms/javascript/usage/#capturing-messages

Still, uncertain whether this is the right place for it.

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts [hunk 2]
```diff
@@ -27,7 +28,7 @@ const swrOptions: SWRConfiguration = {
   revalidateOnFocus: false,
 }
 
-const NETWORKS_WITHOUT_RESTRICTIONS = [SupportedChainId.SEPOLIA]
+const NETWORKS_WITHOUT_RESTRICTIONS = [SupportedChainId.SEPOLIA, SupportedChainId.GNOSIS_CHAIN]
```
Comment (anxolin) [inline libs/tokens/src/updaters/TokensListsUpdater/index.ts:31]: I think you might want to use `fix/disable-token-restrictions-on-gnosis` as base branch instead

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts (hunk 1 above)
Comment (anxolin) [reply libs/tokens/src/updaters/TokensListsUpdater/index.ts:101]: what i had in mind was more to know when this endpoint is not working, so to get a notification if the `catch` is called, not the `then`

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts [hunk 3]
```diff
@@ -80,12 +81,24 @@ export function TokensListsUpdater({ chainId: currentChainId }: TokensListsUpdat
...
+      .then(({ country, ip }) => {
         const isUsUser = country === 'US'
 
         if (isUsUser) {
           setEnvironment({ useCuratedListOnly: true })
           updateLastUpdateTime({ [chainId]: 0 })
+
+          const sentryError = Object.assign(new Error(), {
+            message: '',
+            name: 'USUserError',
```
Comment (anxolin) [inline libs/tokens/src/updaters/TokensListsUpdater/index.ts:93]: I would not add `US` references

Comment (shoom3301) [reply libs/tokens/src/updaters/TokensListsUpdater/index.ts]: Fixed

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts (hunk 2 above)
Comment (shoom3301) [reply libs/tokens/src/updaters/TokensListsUpdater/index.ts]: Fixed

**Comment by shoom3301** [comment]
@anxolin @alfetopito thanks! FIxed

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts (hunk 1 above)
Comment (shoom3301) [reply libs/tokens/src/updaters/TokensListsUpdater/index.ts]: Fixed

**Review by alfetopito — APPROVED** [review]

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts (hunk 2 above)
Comment (anxolin) [reply libs/tokens/src/updaters/TokensListsUpdater/index.ts]: Thanks!

File: libs/tokens/src/updaters/TokensListsUpdater/index.ts (hunk 1 above)
Comment (anxolin) [reply libs/tokens/src/updaters/TokensListsUpdater/index.ts]: Looks great now! did you setup an alert in Sentry yet?

**Comment by anxolin** [comment]
Approve!

**Pull request merged**